server/routes/contacts/manage/contact-details/contactDetailsController.ts

```diff
@@ -78,4 +102,36 @@ export default class ContactDetailsController implements PageHandler {
     }
     return formatNameFirstNameFirst(contact, { customTitle: titleDescription })
   }
+
+  private enrichContactRestrictions(
+    restrictions: ContactRestrictionDetails[],
+    prisonerNumber: string,
+    contactId: string,
+    prisonerContactId: string,
+    restrictionType: string = `PRISONER_CONTACT`,
+  ): ContactRestrictionDetails[] {
+    return restrictions.map(restriction => {
+      const manageRestrictionUrl = this.generateManagePrisonerRestrictionUrl(
+        prisonerNumber,
+        contactId,
+        prisonerContactId,
+        restriction.contactRestrictionId,
+        restrictionType,
+      )
+      return {
+        ...restriction,
+        manageRestrictionUrl,
+      }
+    })
+  }
+
+  private generateManagePrisonerRestrictionUrl(
+    prisonerNumber: string,
+    contactId: string,
+    prisonerContactId: string,
+    restrictionId: number,
+    restrictionType: string = `PRISONER_CONTACT`,
+  ): string {
+    return `/prisoner/${prisonerNumber}/contacts/${contactId}/relationship/${prisonerContactId}/restriction/update/${restrictionType}/enter-restriction/${restrictionId}?returnUrl=/prisoner/${prisonerNumber}/contacts/manage/${contactId}/relationship/${prisonerContactId}`
+  }
```

Comment:
We just do this in the template elsewhere, how come you put it in the controller? Not a huge deal either way just wondering

Reply:
moved to template